tests/contrib/django/test_middleware.py

```diff
@@ -63,6 +63,21 @@ def test_middleware_trace_function_based_view(self):
         eq_(span.get_tag('http.url'), '/fn-view/')
         eq_(span.resource, 'tests.contrib.django.app.views.function_view')
 
+    def test_middleware_trace_error_500(self):
+        # ensures that all error 500 are well traced
+        url = reverse('error-500')
+        response = self.client.get(url)
+        eq_(response.status_code, 500)
+
+        # check for spans
+        spans = self.tracer.writer.pop()
+        eq_(len(spans), 1)
+        span = spans[0]
+        eq_(span.get_tag('http.status_code'), '500')
+        eq_(span.get_tag('http.url'), '/error-500/')
+        eq_(span.resource, 'tests.contrib.django.app.views.error_500')
+        assert "ZeroDivisionError: integer division or modulo by zero" in span.get_tag('error.stack')
```

Comment:
I didn't thought about this case, so yeah, my solution is not working in that case! I'll write a new unit test for the case a middleware throws an error and try to solve it.